core/camel-core-model/src/main/java/org/apache/camel/model/ThrottleDefinition.java

```diff
@@ -57,29 +57,60 @@ public class ThrottleDefinition extends ExpressionNode implements ExecutorServic
     @XmlAttribute
     @Metadata(label = "advanced", javaType = "java.lang.Boolean")
     private String rejectExecution;
+    @XmlAttribute
+    @Metadata(defaultValue = "1000", javaType = "java.time.Duration")
+    private String timePeriodMillis;
+
+    @XmlAttribute
+    @Metadata(label = "advanced", javaType = "org.apache.camel.model.ThrottlingMode", defaultValue = "TotalRequests",
```

Comment:
I dont think it should be advanced, and I think this should be the first option, eg move to top so user select mode first.

Reply:
Before or after `correlationExpression`. It seems like we enforce a particular order via `@XmlType`:

`@XmlType(propOrder = { "expression", "correlationExpression" })` 

I'm wondering if I should do something like:

`@XmlType(propOrder = { "expression", "mode", "correlationExpression" })`